.github/workflows/check-for-non-releasable-actions.yaml

```diff
@@ -0,0 +1,71 @@
+name: Check for non-releasable actions
+on:
+  pull_request:
+    paths:
+      - actions/
+    types:
+      - edited
+      - opened
+      - ready_for_review
+      - synchronize
+
+  push:
+    branches:
+      - main
+    paths:
+      - actions/
+
+jobs:
+  check-for-non-releasable-actions:
+    permissions:
+      contents: read
+      id-token: write
+    runs-on: ubuntu-latest
+    timeout-minutes: 5
+
+    steps:
+      - name: Checkout Code
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+        with:
+          sparse-checkout: |
+            ./actions
+            ./release-please-config.json
+
+      - name: List folders
+        id: list-folders
+        uses: Drafteame/list-folders-action@2138be65c1ec39ea5357a0d47263f330d28b4b02 # v1.0.0
+        with:
+          paths: |
+            ./actions
+
+      - name: Install jsonnet
+        run: go install github.com/google/go-jsonnet/cmd/jsonnet@latest
+
+      - name: Compare config JSON with directories
```

Comment:
this needs some additional modifications since the two arrays aren't in the same shape (needs sed, etc).
I like the approach but I think having a simple script like the one I provided is as easy to follow

Reply:
Personally, I think going with something like `github-script` to implement that thing in JavaScript would be the most readable for me. But that's just me :) 

```
import { readFile } from 'node:fs/promises';

const decoder = new TextDecoder('utf-8');
const releasePleaseConfig = JSON.parse(decoder.decode(await readFile('release-please-config.json')));

const configuredPackageNames = new Set(Object.keys(releasePleaseConfig.packages));
const packageNames = new Set();

// github-script: use glob
// https://github.com/actions/toolkit/tree/main/packages/glob
for await (const folder of Deno.readDir('actions')) {
    if (folder.isDirectory) {
        packageNames.add('actions/' + folder.name);
    }
}

const missingConfigurations = packageNames.difference(configuredPackageNames);
if (missingConfigurations) {
    console.log('The following actions are missing from the release-please-config.json file and thus won\'t be automatically released:');
    console.log(missingConfigurations);
    console.log('Please add them in release-please-config.json!');
} else {
    console.log('All actions are releasable!');
}

```

This also includes the functionality from the previous step.